interfaces/cython/cantera/reactor.pyx

```
@@ -1453,3 +1466,54 @@ cdef class ReactorNet:
 
     def __copy__(self):
         raise NotImplementedError('ReactorNet object is not copyable')
+
+    property preconditioner:
+        """Preconditioner associated with integrator"""
+        def __set__(self, PreconditionerBase precon):
+            # set preconditioner
+            self.net.setPreconditioner(deref(precon.pbase))
+            # set problem type as default of preconditioner
+            self.lin_solver_type = precon.precon_prob_type
+
+    property lin_solver_type:
+        """Associated linear solver type"""
+        def __set__(self, lin_solver_type):
+            if (lin_solver_type == "GMRES"):
+                self.net.setLinSolverType(CxxGMRES)
+            elif (lin_solver_type == "DIAG"):
+                self.net.setLinSolverType(CxxDIAG)
+            elif (lin_solver_type == "BAND + NOJAC"):
+                self.net.setLinSolverType(CxxBAND + CxxNOJAC)
+            else:
+                self.net.setLinSolverType(CxxDENSE + CxxNOJAC)
+                if (lin_solver_type != "DENSE + NOJAC"):
+                    warnings.warn("Problem type not found, set to \"DENSE + NOJAC\"")
```

I think the last case here should raise an exception. Also, in Python parentheses around the condition of an `if` statement should not be used, unless the condition extends across multiple lines.

This may be moot depending on how you decide to refactor the implementation of the C++ `setLinearSolverType` method.